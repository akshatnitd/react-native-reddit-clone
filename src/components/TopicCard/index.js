import React from "react"
import PropTypes from "prop-types"
import { View, Text } from "react-native"
import { colorStyle, fontStyle } from "../../style"

import Card from "../Card"
import Button from "../Button"
import Icon from 'react-native-vector-icons/MaterialIcons'

class TopicCard extends React.Component {
  
  handleUpvoteTopic = (topicId) => {
    this.props.actions.upvoteTopic(topicId)
  }

  handleDownvoteTopic = (topicId) => {
    this.props.actions.downvoteTopic(topicId)
  }

  render() {
    const { data, actions: { downvoteTopic, upvoteTopic } } = this.props

    return (
      <View>
        <Card containerStyle={{ marginHorizontal: 0, marginVertical: 3 }}>
          <View
            style={{
              paddingTop: 10,
              paddingHorizontal: 15,
              flexDirection: "row",
              alignItems: "center",
            }}
          >
            <Text style={fontStyle.caption}>@{data.username}</Text>
          </View>

          <View style={{ paddingVertical: 3, paddingHorizontal: 15 }}>
            <Text style={fontStyle.body2}>{data.title}</Text>
          </View>

          <View style={{ flexDirection: "row", paddingTop: 5 }}>
            
            <Icon.Button 
              name="thumb-up" 
              rippleColor={colorStyle.primaryDark}
              backgroundColor="#f2f2f2" 
              onPress={this.handleUpvoteTopic.bind(this, data.id)} 
              size={30} 
              color="#0f0" 
            >
            </Icon.Button>
            
            <Icon.Button 
              name="thumb-down" 
              backgroundColor="#f2f2f2"
              onPress={this.handleUpvoteTopic.bind(this, data.id)} 
              size={30} 
              color="#f00" 
            >
            </Icon.Button>

            
            <Text
              style={{
                paddingVertical: 8,
                paddingHorizontal: 3
              }}
            >
              {data.vote} votes
            </Text>

          </View>

        </Card>
      </View>
    )
  }
}

TopicCard.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.string.isRequired,
    username: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    vote: PropTypes.number.isRequired,
  }).isRequired,
  actions: PropTypes.shape({
    downvoteTopic: PropTypes.func.isRequired,
    upvoteTopic: PropTypes.func.isRequired,
  }).isRequired,
}

export default TopicCard
