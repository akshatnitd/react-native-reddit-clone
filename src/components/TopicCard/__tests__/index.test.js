import React from "react"
import renderer from "react-test-renderer"

import TopicCard from ".."

it("renders correctly", () => {
  const downvoteTopicAction = jest.fn()
  const upvoteTopicAction = jest.fn()
  const tree = renderer.create(
    <TopicCard
      data={{
        id: "5a53344f82e7eec2a93dae9b",
        username: "Eva",
        title:
          "id ad nulla mollit incididunt exercitation ipsum quis ullamco nostrud",
        vote: 33,
        createdAt: "Tue Jan 16 2018 15:43:13 GMT+0000 (UTC)",
        modifiedAt: "Wed Jan 03 2018 03:28:48 GMT+0000 (UTC)",
      }}
      actions={{
        downvoteTopic: downvoteTopicAction,
        upvoteTopic: upvoteTopicAction,
      }}
    />,
  )
  expect(tree.toJSON()).toMatchSnapshot()

  const instance = tree.getInstance()
  instance.handleUpvoteTopic("asdf")
  expect(upvoteTopicAction.mock.calls).toHaveLength(1)

  instance.handleDownvoteTopic("asdf")
  expect(upvoteTopicAction.mock.calls).toHaveLength(1)
})
