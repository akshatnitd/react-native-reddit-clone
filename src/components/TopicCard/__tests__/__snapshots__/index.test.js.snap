// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    containerStyle={
      Object {
        "marginHorizontal": 0,
        "marginVertical": 3,
      }
    }
    style={
      Array [
        Object {
          "backgroundColor": "#f2f2f2",
          "elevation": 1,
          "flex": 1,
          "flexWrap": "nowrap",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        },
        Object {
          "marginHorizontal": 0,
          "marginVertical": 3,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingHorizontal": 15,
          "paddingTop": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "rgba(0,0,0,0.54)",
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 16,
          }
        }
      >
        @
        Eva
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 15,
          "paddingVertical": 3,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "rgba(0,0,0,0.87)",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "600",
            "letterSpacing": -0.154,
            "lineHeight": 24,
          }
        }
      >
        id ad nulla mollit incididunt exercitation ipsum quis ullamco nostrud
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingTop": 5,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 6,
              "borderWidth": 0,
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            onPress={[Function]}
            rippleColor="#0099cc"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "borderColor": "#bdbdbd",
                  "borderRadius": 6,
                  "flex": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 3,
                  "paddingVertical": 8,
                },
                undefined,
              ]
            }
            textStyle={
              Object {
                "color": "#0099cc",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(0,0,0,0.87)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "600",
                    "letterSpacing": -0.154,
                    "lineHeight": 20,
                  },
                  Object {
                    "color": "#0099cc",
                  },
                ]
              }
            >
              UPVOTE
            </Text>
          </View>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "paddingHorizontal": 3,
            "paddingVertical": 8,
          }
        }
      >
        33
         votes
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 6,
              "borderWidth": 0,
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            onPress={[Function]}
            rippleColor="#ff4444"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "borderColor": "#bdbdbd",
                  "borderRadius": 6,
                  "flex": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 3,
                  "paddingVertical": 8,
                },
                undefined,
              ]
            }
            textStyle={
              Object {
                "color": "#ff4444",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(0,0,0,0.87)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "600",
                    "letterSpacing": -0.154,
                    "lineHeight": 20,
                  },
                  Object {
                    "color": "#ff4444",
                  },
                ]
              }
            >
              DOWNVOTE
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
